container/container_unix.go

```diff
@@ -241,8 +241,11 @@ func (container *Container) SecretMounts() ([]Mount, error) {
 			return nil, err
 		}
 		mounts = append(mounts, Mount{
-			Source:      fPath,
-			Destination: r.File.Name,
+			Source: fPath,
+			// runc >= 1.0.0-rc94 wants Destination to be absolute,
+			// so we prepend "/" here.
+			// https://github.com/opencontainers/runc/issues/2928
+			Destination: "/" + r.File.Name,
```

Comment:
Can you move this commit to a separate PR?

This should also probably take the same approach as  is used for secrets; https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/container/container_unix.go#L234

And use a function like `getSecretTargetPath()`; https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/container/container.go#L711-L717

Difference would be that `containerSecretMountPath` would be `/` on Linux, and `C:/` on Windows (?))

https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/container/container_windows.go#L15

https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/container/container_unix.go#L30

Reply:
Opened https://github.com/moby/moby/pull/42370 with the above